godot_project/editor/rendering/atomic_structure_renderer/representation/stick_representation/stick_representation.gd

```diff
@@ -584,10 +583,13 @@ static func _calc_up_vector_for_higher_bond(in_first_atom_id: int, in_second_ato
 	return plane.normal
 
 
-func set_partially_selected_bonds(in_partially_selected_bonds: PackedInt32Array) -> void:
-	_refresh_bond_partial_influence_status(_current_bond_partial_selection.keys())
-	_current_bond_partial_selection.clear()
-	_refresh_bond_partial_influence_status(in_partially_selected_bonds)
+func refresh_bond_influence(in_partially_selected_bonds: PackedInt32Array) -> void:
+	var bonds_to_refresh: Dictionary = {}
+	for bond_id: int in in_partially_selected_bonds:
+		bonds_to_refresh[bond_id] = true
+	for bond_id: int in _current_bond_partial_selection:
+		bonds_to_refresh[bond_id] = true
+	_refresh_bond_partial_influence_status(bonds_to_refresh.keys())
```

Comment:
Those changes do not fix anything, this change have only one goal in mind, to avoid potential call of _refresh_bond_partial_influence_status logic twice for the same bond